src/main/scala/com.snowplowanalytics.snowplow.badrows/Failure.scala

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2018-2019 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.snowplow.badrows
+
+import java.time.Instant
+
+import cats.data.NonEmptyList
+import cats.syntax.functor._
+import com.snowplowanalytics.iglu.client.ClientError
+import com.snowplowanalytics.iglu.core.{SchemaCriterion, SchemaKey}
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.auto._
+import io.circe.java8.time._
+import io.circe.syntax._
+
+sealed trait Failure
+object Failure {
+  implicit val encodeFailure: Encoder[Failure] = Encoder.instance {
```

Comment:
Isn't it equal to `deriveEncoder[Failure]`?

Reply:
I don't think semiauto works with ADT but I might be wrong